address warnings in resource/resource_gen.cpp etc, then revert workaround

As discussed in #298, g++ 5.4.0 issues some new warnings that must be disabled in order to be non-fatal due to `-Werror`.  In order to support this compiler, several `-Wno-*` options were added to resources/Makefile.am (8a4b016c4b39938cf8a3041d8b006b163d02c3cc).   However, one that is needed, `-Wno-maybe-uninitialized`, could not be added because it is not understood by clang 3.8 in travis, and an unknown warning option is fatal.

The workaround (68569905a21592f290bf581a78e270336df450cd) was to add `-Wno-error` to resources/Makefile.am, but that should only be a temporary solution until the root cause of the warnings is found.

Here is the warning output:
```
jimbo:~/proj/flux-sched/resource > make
Making all in planner
make[1]: Entering directory '/home/garlick/proj/flux-sched/resource/planner'
Making all in .
make[2]: Entering directory '/home/garlick/proj/flux-sched/resource/planner'
  CC       libplanner_la-planner.lo
  CCLD     libplanner.la
ar: `u' modifier ignored since `D' is the default (see `U')
make[2]: Leaving directory '/home/garlick/proj/flux-sched/resource/planner'
Making all in test
make[2]: Entering directory '/home/garlick/proj/flux-sched/resource/planner/test'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/garlick/proj/flux-sched/resource/planner/test'
make[1]: Leaving directory '/home/garlick/proj/flux-sched/resource/planner'
Making all in .
make[1]: Entering directory '/home/garlick/proj/flux-sched/resource'
  CXX      utilities/utilities_grug2dot-grug2dot.o
  CXX      utilities_grug2dot-resource_gen_spec.o
In file included from /opt/boost-1.53/include/boost/graph/detail/adjacency_list.hpp:33:0,
                 from /opt/boost-1.53/include/boost/graph/adjacency_list.hpp:246,
                 from resource_gen_spec.cpp:26:
/opt/boost-1.53/include/boost/graph/detail/adj_list_edge_iterator.hpp: In member function ‘void boost::vec_adj_list_impl<Graph, Config, Base>::copy_impl(const boost::vec_adj_list_impl<Graph, Config, Base>&) [with Graph = boost::adjacency_li/opt/boost-1.53/include/boost/graph/detail/adj_list_edge_iterator.hpp:95:84: warning: ‘*((void*)& ei_end +32)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
               && edges BOOST_GRAPH_MEMBER first != x.edges BOOST_GRAPH_MEMBER first);
                                                                                    ^
In file included from /opt/boost-1.53/include/boost/graph/adjacency_list.hpp:246:0,
                 from resource_gen_spec.cpp:26:
/opt/boost-1.53/include/boost/graph/detail/adjacency_list.hpp:2119:27: note: ‘*((void*)& ei_end +32)’ was declared here
         edge_iterator ei, ei_end;
                           ^
In file included from /opt/boost-1.53/include/boost/graph/detail/adjacency_list.hpp:33:0,
                 from /opt/boost-1.53/include/boost/graph/adjacency_list.hpp:246,
                 from resource_gen_spec.cpp:26:
/opt/boost-1.53/include/boost/graph/detail/adj_list_edge_iterator.hpp:70:9: warning: ‘*((void*)& ei +48)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         if (edges BOOST_GRAPH_MEMBER first == edges BOOST_GRAPH_MEMBER second) 
         ^
In file included from /opt/boost-1.53/include/boost/graph/adjacency_list.hpp:246:0,
                 from resource_gen_spec.cpp:26:
/opt/boost-1.53/include/boost/graph/detail/adjacency_list.hpp:2119:23: note: ‘*((void*)& ei +48)’ was declared here
         edge_iterator ei, ei_end;
                       ^
  CXXLD    utilities/grug2dot
  CXX      utilities/utilities_resource_query-resource-query.o
  CXX      utilities/utilities_resource_query-command.o
  CXX      utilities_resource_query-dfu_traverse.o
  CXX      utilities_resource_query-dfu_traverse_impl.o
  CXX      utilities_resource_query-resource_gen.o
In file included from /opt/boost-1.53/include/boost/graph/detail/adjacency_list.hpp:33:0,
                 from /opt/boost-1.53/include/boost/graph/adjacency_list.hpp:246,
                 from resource_graph.hpp:29,
                 from resource_gen.hpp:30,
                 from resource_gen.cpp:30:
/opt/boost-1.53/include/boost/graph/detail/adj_list_edge_iterator.hpp: In member function ‘void boost::vec_adj_list_impl<Graph, Config, Base>::copy_impl(const boost::vec_adj_list_impl<Graph, Config, Base>&) [with Graph = boost::adjacency_list<boost::vecS, boost::vecS, boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t>; Config = boost::detail::adj_list_gen<boost::adjacency_list<boost::vecS, boost::vecS, boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t>, boost::vecS, boost::vecS, boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t, boost::no_property, boost::listS>::config; Base = boost::directed_graph_helper<boost::detail::adj_list_gen<boost::adjacency_list<boost::vecS, boost::vecS, boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t>, boost::vecS, boost::vecS, boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t, boost::no_property, boost::listS>::config>]’:
/opt/boost-1.53/include/boost/graph/detail/adj_list_edge_iterator.hpp:70:9: warning: ‘*((void*)& ei +48)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         if (edges BOOST_GRAPH_MEMBER first == edges BOOST_GRAPH_MEMBER second) 
         ^
In file included from /opt/boost-1.53/include/boost/graph/adjacency_list.hpp:246:0,
                 from resource_graph.hpp:29,
                 from resource_gen.hpp:30,
                 from resource_gen.cpp:30:
/opt/boost-1.53/include/boost/graph/detail/adjacency_list.hpp:2119:23: note: ‘*((void*)& ei +48)’ was declared here
         edge_iterator ei, ei_end;
                       ^
  CXX      utilities_resource_query-resource_gen_spec.o
In file included from /opt/boost-1.53/include/boost/graph/detail/adjacency_list.hpp:33:0,
                 from /opt/boost-1.53/include/boost/graph/adjacency_list.hpp:246,
                 from resource_gen_spec.cpp:26:
/opt/boost-1.53/include/boost/graph/detail/adj_list_edge_iterator.hpp: In member function ‘void boost::vec_adj_list_impl<Graph, Config, Base>::copy_impl(const boost::vec_adj_list_impl<Graph, Config, Base>&) [with Graph = boost::adjacency_list<boost::vecS, boost::vecS, boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t>; Config = boost::detail::adj_list_gen<boost::adjacency_list<boost::vecS, boost::vecS, boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t>, boost::vecS, boost::vecS, boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t, boost::no_property, boost::listS>::config; Base = boost::directed_graph_helper<boost::detail::adj_list_gen<boost::adjacency_list<boost::vecS, boost::vecS, boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t>, boost::vecS, boost::vecS, boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t, boost::no_property, boost::listS>::config>]’:
/opt/boost-1.53/include/boost/graph/detail/adj_list_edge_iterator.hpp:95:84: warning: ‘*((void*)& ei_end +32)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
               && edges BOOST_GRAPH_MEMBER first != x.edges BOOST_GRAPH_MEMBER first);
                                                                                    ^
In file included from /opt/boost-1.53/include/boost/graph/adjacency_list.hpp:246:0,
                 from resource_gen_spec.cpp:26:
/opt/boost-1.53/include/boost/graph/detail/adjacency_list.hpp:2119:27: note: ‘*((void*)& ei_end +32)’ was declared here
         edge_iterator ei, ei_end;
                           ^
In file included from /opt/boost-1.53/include/boost/graph/detail/adjacency_list.hpp:33:0,
                 from /opt/boost-1.53/include/boost/graph/adjacency_list.hpp:246,
                 from resource_gen_spec.cpp:26:
/opt/boost-1.53/include/boost/graph/detail/adj_list_edge_iterator.hpp:70:9: warning: ‘*((void*)& ei +48)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         if (edges BOOST_GRAPH_MEMBER first == edges BOOST_GRAPH_MEMBER second) 
         ^
In file included from /opt/boost-1.53/include/boost/graph/adjacency_list.hpp:246:0,
                 from resource_gen_spec.cpp:26:
/opt/boost-1.53/include/boost/graph/detail/adjacency_list.hpp:2119:23: note: ‘*((void*)& ei +48)’ was declared here
         edge_iterator ei, ei_end;
                       ^
  CXXLD    utilities/resource-query
make[1]: Leaving directory '/home/garlick/proj/flux-sched/resource'
```

Another warning that is popping up in that code on a system with gcc 4.9.2
```
  CXX      utilities_resource_query-resource_gen.o
resource_gen.cpp: In member function ‘int dfs_emitter_t::path_prefix(const string&, int, std::string&)’:
resource_gen.cpp:90:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     while (occurrence != (num_slashes - uplevel + 1)) {
                       ^
```

I built the entire GCC-5.4 stack from the source on my centos7 container to test this out. But with my side-installed GCC

```
bash-4.2$ g++ --version
g++ (GCC) 5.4.0
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

I am getting: 

```
make[3]: Entering directory `/usr/src/resource/libjobspec'
/bin/sh ../../libtool  --tag=CXX   --mode=link g++ -std=c++11  -g -O2   -o flux-jobspec-validate flux-jobspec-validate.o libjobspec_conv.la -lyaml-cpp
libtool: link: g++ -std=c++11 -g -O2 -o flux-jobspec-validate flux-jobspec-validate.o  ./.libs/libjobspec_conv.a /usr/src/GCC/lib/../lib64/libstdc++.so -lyaml-cpp -Wl,-rpath -Wl,/usr/src/GCC/lib/../lib64 -Wl,-rpath -Wl,/usr/src/GCC/lib/../lib64
./.libs/libjobspec_conv.a(libjobspec_conv_la-jobspec.o): In function `YAML::Node::Scalar[abi:cxx11]() const':
/usr/include/yaml-cpp/node/impl.h:158: undefined reference to `YAML::detail::node_data::empty_scalar[abi:cxx11]'
./.libs/libjobspec_conv.a(libjobspec_conv_la-jobspec.o): In function `Flux::Jobspec::Jobspec::Jobspec(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/usr/src/resource/libjobspec/jobspec.cpp:425: undefined reference to `YAML::Load(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
./.libs/libjobspec_conv.a(libjobspec_conv_la-jobspec.o): In function `YAML::Node::Scalar[abi:cxx11]() const':
/usr/include/yaml-cpp/node/impl.h:158: undefined reference to `YAML::detail::node_data::empty_scalar[abi:cxx11]'
/usr/include/yaml-cpp/node/impl.h:158: undefined reference to `YAML::detail::node_data::empty_scalar[abi:cxx11]'
collect2: error: ld returned 1 exit status
make[3]: *** [flux-jobspec-validate] Error 1
make[3]: Leaving directory `/usr/src/resource/libjobspec'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/usr/src/resource'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/usr/src'
make: *** [all] Error 2
```

This seems to be a C++ API mismatch between GCC 5.4 and the yaml shared library.

I'm trying to compile the latest flux-sched master with spack on Cori using gcc 7.5.0.  I get the error included below.  Presumably this is a boost error?  I am wondering if the removal of `-Wno-maybe-uninitialized` is what unveiled this error.

```
In file included from /global/u2/h/herbein1/Repositories/spack/opt/spack/cray-sles15-x86_64/gcc-7.5.0/boost-1.73.0-bswt6cvinpmlaovppy67v67c2nfr5j3s/include/boost/graph/detail/adjacency_list.hpp:34:0,
                 from /global/u2/h/herbein1/Repositories/spack/opt/spack/cray-sles15-x86_64/gcc-7.5.0/boost-1.73.0-bswt6cvinpmlaovppy67v67c2nfr5j3s/include/boost/graph/adjacency_list.hpp:255,
                 from /tmp/herbein1/spack-stage/spack-stage-flux-sched-master-bhchkv7zuqg7tipvln3swpji5s3d4ubx/spack-src/resource/readers/resource_spec_grug.cpp:26:
/global/u2/h/herbein1/Repositories/spack/opt/spack/cray-sles15-x86_64/gcc-7.5.0/boost-1.73.0-bswt6cvinpmlaovppy67v67c2nfr5j3s/include/boost/graph/detail/adj_list_edge_iterator.hpp: In member function 'void boost::vec_adj_list_impl<Graph
, Config, Base>::copy_impl(const boost::vec_adj_list_impl<Graph, Config, Base>&) [with Graph = boost::adjacency_list<boost::vecS, boost::vecS, boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_g
en_t>; Config = boost::detail::adj_list_gen<boost::adjacency_list<boost::vecS, boost::vecS, boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t>, boost::vecS, boost::vecS, boost::directedS,
Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t, boost::no_property, boost::listS>::config; Base = boost::directed_graph_helper<boost::detail::adj_list_gen<boost::adjacency_list<boost::vecS, boost::vecS,
boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t>, boost::vecS, boost::vecS, boost::directedS, Flux::resource_model::resource_pool_gen_t, Flux::resource_model::relation_gen_t, boost::no_p
roperty, boost::listS>::config>]':
/global/u2/h/herbein1/Repositories/spack/opt/spack/cray-sles15-x86_64/gcc-7.5.0/boost-1.73.0-bswt6cvinpmlaovppy67v67c2nfr5j3s/include/boost/graph/detail/adj_list_edge_iterator.hpp:112:21: error: '*((void*)& ei_end +32)' may be used unin
itialized in this function [-Werror=maybe-uninitialized]
                 || (vCurr != vEnd
                    ~~~~~~~~~~~~~~
                     && edges BOOST_GRAPH_MEMBER first
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                         != x.edges BOOST_GRAPH_MEMBER first);
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /global/u2/h/herbein1/Repositories/spack/opt/spack/cray-sles15-x86_64/gcc-7.5.0/boost-1.73.0-bswt6cvinpmlaovppy67v67c2nfr5j3s/include/boost/graph/adjacency_list.hpp:255:0,
                 from /tmp/herbein1/spack-stage/spack-stage-flux-sched-master-bhchkv7zuqg7tipvln3swpji5s3d4ubx/spack-src/resource/readers/resource_spec_grug.cpp:26:
/global/u2/h/herbein1/Repositories/spack/opt/spack/cray-sles15-x86_64/gcc-7.5.0/boost-1.73.0-bswt6cvinpmlaovppy67v67c2nfr5j3s/include/boost/graph/detail/adjacency_list.hpp:2186:27: note: '*((void*)& ei_end +32)' was declared here
         edge_iterator ei, ei_end;
                           ^~~~~~
In file included from /global/u2/h/herbein1/Repositories/spack/opt/spack/cray-sles15-x86_64/gcc-7.5.0/boost-1.73.0-bswt6cvinpmlaovppy67v67c2nfr5j3s/include/boost/graph/detail/adjacency_list.hpp:34:0,
                 from /global/u2/h/herbein1/Repositories/spack/opt/spack/cray-sles15-x86_64/gcc-7.5.0/boost-1.73.0-bswt6cvinpmlaovppy67v67c2nfr5j3s/include/boost/graph/adjacency_list.hpp:255,
                 from /tmp/herbein1/spack-stage/spack-stage-flux-sched-master-bhchkv7zuqg7tipvln3swpji5s3d4ubx/spack-src/resource/readers/resource_spec_grug.cpp:26:
/global/u2/h/herbein1/Repositories/spack/opt/spack/cray-sles15-x86_64/gcc-7.5.0/boost-1.73.0-bswt6cvinpmlaovppy67v67c2nfr5j3s/include/boost/graph/detail/adj_list_edge_iterator.hpp:80:13: error: '*((void*)& ei +48)' may be used uninitial
ized in this function [-Werror=maybe-uninitialized]
             if (edges BOOST_GRAPH_MEMBER first
             ^~
In file included from /global/u2/h/herbein1/Repositories/spack/opt/spack/cray-sles15-x86_64/gcc-7.5.0/boost-1.73.0-bswt6cvinpmlaovppy67v67c2nfr5j3s/include/boost/graph/adjacency_list.hpp:255:0,
                 from /tmp/herbein1/spack-stage/spack-stage-flux-sched-master-bhchkv7zuqg7tipvln3swpji5s3d4ubx/spack-src/resource/readers/resource_spec_grug.cpp:26:
/global/u2/h/herbein1/Repositories/spack/opt/spack/cray-sles15-x86_64/gcc-7.5.0/boost-1.73.0-bswt6cvinpmlaovppy67v67c2nfr5j3s/include/boost/graph/detail/adjacency_list.hpp:2186:23: note: '*((void*)& ei +48)' was declared here
         edge_iterator ei, ei_end;
                       ^~
/global/u2/h/herbein1/Repositories/spack/opt/spack/cray-sles15-x86_64/gcc-7.5.0/boost-1.73.0-bswt6cvinpmlaovppy67v67c2nfr5j3s/include/boost/graph/detail/adjacency_list.hpp:2186:23: error: '*((void*)& ei +40)' may be used uninitialized i
n this function [-Werror=maybe-uninitialized]
/global/u2/h/herbein1/Repositories/spack/opt/spack/cray-sles15-x86_64/gcc-7.5.0/boost-1.73.0-bswt6cvinpmlaovppy67v67c2nfr5j3s/include/boost/graph/detail/adjacency_list.hpp:2186:23: error: '*((void*)(& ei)+32).__gnu_cxx::__normal_iterato
r<boost::detail::stored_edge_property<long unsigned int, Flux::resource_model::relation_gen_t>*, std::vector<boost::detail::stored_edge_property<long unsigned int, Flux::resource_model::relation_gen_t>, std::allocator<boost::detail::sto
red_edge_property<long unsigned int, Flux::resource_model::relation_gen_t> > > >::_M_current' may be used uninitialized in this function [-Werror=maybe-uninitialized]
cc1plus: all warnings being treated as errors
```

--- a/config/ax_boost_base.m4
+++ b/config/ax_boost_base.m4
@@ -133,7 +133,7 @@ AC_DEFUN([_AX_BOOST_BASE_RUNDETECT],[
         AC_MSG_CHECKING([for boostlib >= $1 ($WANT_BOOST_VERSION) includes in "$_AX_BOOST_BASE_boost_path/include"])
          AS_IF([test -d "$_AX_BOOST_BASE_boost_path/include" && test -r "$_AX_BOOST_BASE_boost_path/include"],[
            AC_MSG_RESULT([yes])
-           BOOST_CPPFLAGS="-I$_AX_BOOST_BASE_boost_path/include"
+           BOOST_CPPFLAGS="-isystem $_AX_BOOST_BASE_boost_path/include"
            for _AX_BOOST_BASE_boost_path_tmp in $multiarch_libsubdir $libsubdirs; do
                 AC_MSG_CHECKING([for boostlib >= $1 ($WANT_BOOST_VERSION) lib path in "$_AX_BOOST_BASE_boost_path/$_AX_BOOST_BASE_boost_path_tmp"])
                 AS_IF([test -d "$_AX_BOOST_BASE_boost_path/$_AX_BOOST_BASE_boost_path_tmp" && test -r "$_AX_BOOST_BASE_boost_path/$_AX_BOOST_BASE_boost_path_tmp" ],[
@@ -156,7 +156,7 @@ AC_DEFUN([_AX_BOOST_BASE_RUNDETECT],[
                     if ls "$_AX_BOOST_BASE_boost_path_tmp/$libsubdir/libboost_"* >/dev/null 2>&1 ; then break; fi
                 done
                 BOOST_LDFLAGS="-L$_AX_BOOST_BASE_boost_path_tmp/$libsubdir"
-                BOOST_CPPFLAGS="-I$_AX_BOOST_BASE_boost_path_tmp/include"
+                BOOST_CPPFLAGS="-isystem $_AX_BOOST_BASE_boost_path_tmp/include"
                 break;
             fi
         done
@@ -207,12 +207,12 @@ AC_DEFUN([_AX_BOOST_BASE_RUNDETECT],[
                         _version=$_version_tmp
                     fi
                     VERSION_UNDERSCORE=`echo $_version | sed 's/\./_/'`
-                    BOOST_CPPFLAGS="-I$_AX_BOOST_BASE_boost_path/include/boost-$VERSION_UNDERSCORE"
+                    BOOST_CPPFLAGS="-isystem $_AX_BOOST_BASE_boost_path/include/boost-$VERSION_UNDERSCORE"
                 done
                 dnl if nothing found search for layout used in Windows distributions
                 if test -z "$BOOST_CPPFLAGS"; then
                     if test -d "$_AX_BOOST_BASE_boost_path/boost" && test -r "$_AX_BOOST_BASE_boost_path/boost"; then
-                        BOOST_CPPFLAGS="-I$_AX_BOOST_BASE_boost_path"
+                        BOOST_CPPFLAGS="-isystem $_AX_BOOST_BASE_boost_path"
                     fi
                 fi
                 dnl if we found something and BOOST_LDFLAGS was unset before
@@ -240,7 +240,7 @@ AC_DEFUN([_AX_BOOST_BASE_RUNDETECT],[
                 done
 
                 VERSION_UNDERSCORE=`echo $_version | sed 's/\./_/'`
-                BOOST_CPPFLAGS="-I$best_path/include/boost-$VERSION_UNDERSCORE"
+                BOOST_CPPFLAGS="-isystem $best_path/include/boost-$VERSION_UNDERSCORE"
                 if test -z "$_AX_BOOST_BASE_boost_lib_path" ; then
                     for libsubdir in $libsubdirs ; do
                         if ls "$best_path/$libsubdir/libboost_"* >/dev/null 2>&1 ; then break; fi
@@ -260,7 +260,7 @@ AC_DEFUN([_AX_BOOST_BASE_RUNDETECT],[
                     V_CHECK=`expr $stage_version_shorten \>\= $_version`
                     if test "x$V_CHECK" = "x1" && test -z "$_AX_BOOST_BASE_boost_lib_path" ; then
                         AC_MSG_NOTICE(We will use a staged boost library from $BOOST_ROOT)
-                        BOOST_CPPFLAGS="-I$BOOST_ROOT"
+                        BOOST_CPPFLAGS="-isystem $BOOST_ROOT"
                         BOOST_LDFLAGS="-L$BOOST_ROOT/stage/$libsubdir"
                     fi
                 fi
@@ -298,5 +298,12 @@ AC_DEFUN([_AX_BOOST_BASE_RUNDETECT],[
 
     CPPFLAGS="$CPPFLAGS_SAVED"
     LDFLAGS="$LDFLAGS_SAVED"
+
+    dnl Due to a change in GCC 6+, searching a default system path via -isystem
+    dnl causes compilation issues.  Let the compiler find headers in default
+    dnl system paths "naturally".
+    if test "$BOOST_CPPFLAGS" = "-isystem /usr/include" ; then
+       BOOST_CPPFLAGS=""
+    fi
 
 ])

